sway-lib-std/src/call_frames.sw

```diff
@@ -20,13 +20,48 @@ const SECOND_PARAMETER_OFFSET: u64 = 74;
 //  Accessing the current call frame
 //
 /// Get the current contract's id when called in an internal context.
+///
+/// # Additional Information
+///
 /// **_Note:_** If called in an external context, this will **not** return a contract ID.
-// @dev If called externally, will actually return a pointer to the transaction ID.
+/// If called externally, will actually return a pointer to the transaction ID.
+/// 
+/// # Returns
+///
+/// * [ContractId] - The contract id of this contract.
+///
+/// # Examples
+///
+/// ```sway
+/// use std::{call_frames::contract_id, context::this_balance, token::mint_to_contract};
+///
+/// fn foo() {
+///     let this_contract = contract_id();
+///     mint_to_contract(50, this_contract);
```

Comment:
I dont think this works? mint already directly mints to the self contract, so this would transfer from self to self. also just makes it confusing even if it works. Sorry to be annoying about this. Maybe just `mint(50)` and then `Address::from(ZERO_B256).transfer(50, this_contract)`

Reply:
This will need some revisions for the new spec but will work for now